### ISSUE TITLE
Can't connnect to SQL Server running in a different container

### ISSUE THREAD

**Issue by rscarrasco** [liquibase/docker#29]
I have a container running SQL Server, set up by following [Microsoft recomendation](https://docs.microsoft.com/en-us/sql/linux/sql-server-linux-configure-docker?view=sql-server-2017). My intent is to use it as a sandbox environment, to test my changelogs before actually committing them, and as so I'm trying to run the Liquibase image in a different container, connecting to the database, but it fails to connect do the database.

This is how I'm trying to execute it:

```sh
docker run --rm \
           --volume /local/path/to/changelogs:/liquibase/changelog \
           liquibase/liquibase \
           --url='jdbc:sqlserver://0.0.0.0:1433;database=MySandbox;' \
           --changeLogFile=/liquibase/changelog/db-changelog-master.xml \
           --username=sa \
           --password=MyStrongPassword@ \
           --logLevel=debug \
           validate
```
and this is what happens:

```
(a bunch of log messages)
[2020-08-12 14:30:11] FINE [com.microsoft.sqlserver.jdbc.SQLServerException logException] *** SQLException:ConnectionID:1 com.microsoft.sqlserver.jdbc.SQLServerException: The TCP/IP connection to the host 0.0.0.0, port 1433 has failed. Error: "Connection refused (Connection refused). Verify the connection properties. Make sure that an instance of SQL Server is running on the host and accepting TCP/IP connections at the port. Make sure that TCP connections to the port are not blocked by a firewall.". The TCP/IP connection to the host 0.0.0.0, port 1433 has failed. Error: "Connection refused (Connection refused). Verify the connection properties. Make sure that an instance of SQL Server is running on the host and accepting TCP/IP connections at the port. Make sure that TCP connections to the port are not blocked by a firewall.".
Unexpected error running Liquibase: liquibase.exception.DatabaseException: liquibase.exception.DatabaseException: Connection could not be created to jdbc:sqlserver://0.0.0.0:1433;database=MySandbox; with driver com.microsoft.sqlserver.jdbc.SQLServerDriver.  Possibly the wrong driver for the given database URL
For more information, please use the --logLevel flag
[2020-08-12 14:30:11] SEVERE [liquibase.integration] Unexpected error running Liquibase: liquibase.exception.DatabaseException: liquibase.exception.DatabaseException: Connection could not be created to jdbc:sqlserver://0.0.0.0:1433;database=MySandbox; with driver com.microsoft.sqlserver.jdbc.SQLServerDriver.  Possibly the wrong driver for the given database URL
liquibase.exception.DatabaseException: liquibase.exception.DatabaseException: liquibase.exception.DatabaseException: liquibase.exception.DatabaseException: Connection could not be created to jdbc:sqlserver://0.0.0.0:1433;database=MySandbox; with driver com.microsoft.sqlserver.jdbc.SQLServerDriver.  Possibly the wrong driver for the given database URL
    at liquibase.integration.commandline.CommandLineUtils.createDatabaseObject(CommandLineUtils.java:127)
    at liquibase.integration.commandline.Main.doMigration(Main.java:1249)
    at liquibase.integration.commandline.Main$1.lambda$run$0(Main.java:316)
    at liquibase.Scope.lambda$child$0(Scope.java:159)
    at liquibase.Scope.child(Scope.java:170)
    at liquibase.Scope.child(Scope.java:158)
    at liquibase.Scope.child(Scope.java:137)
    at liquibase.Scope.child(Scope.java:183)
    at liquibase.Scope.child(Scope.java:187)
    at liquibase.integration.commandline.Main$1.run(Main.java:315)
    at liquibase.integration.commandline.Main$1.run(Main.java:166)
    at liquibase.Scope.child(Scope.java:170)
    at liquibase.Scope.child(Scope.java:144)
    at liquibase.integration.commandline.Main.run(Main.java:166)
    at liquibase.integration.commandline.Main.main(Main.java:145)
Caused by: liquibase.exception.DatabaseException: liquibase.exception.DatabaseException: liquibase.exception.DatabaseException: Connection could not be created to jdbc:sqlserver://0.0.0.0:1433;database=MySandbox; with driver com.microsoft.sqlserver.jdbc.SQLServerDriver.  Possibly the wrong driver for the given database URL
    at liquibase.database.DatabaseFactory.openConnection(DatabaseFactory.java:250)
    at liquibase.database.DatabaseFactory.openDatabase(DatabaseFactory.java:140)
    at liquibase.integration.commandline.CommandLineUtils.createDatabaseObject(CommandLineUtils.java:92)
    ... 14 more
Caused by: liquibase.exception.DatabaseException: liquibase.exception.DatabaseException: Connection could not be created to jdbc:sqlserver://0.0.0.0:1433;database=MySandbox; with driver com.microsoft.sqlserver.jdbc.SQLServerDriver.  Possibly the wrong driver for the given database URL
    at liquibase.database.ConnectionServiceFactory.create(ConnectionServiceFactory.java:36)
    at liquibase.database.DatabaseFactory.openConnection(DatabaseFactory.java:247)
    ... 16 more
Caused by: liquibase.exception.DatabaseException: Connection could not be created to jdbc:sqlserver://0.0.0.0:1433;database=MySandbox; with driver com.microsoft.sqlserver.jdbc.SQLServerDriver.  Possibly the wrong driver for the given database URL
    at liquibase.database.jvm.JdbcConnection.open(JdbcConnection.java:43)
    at com.datical.liquibase.ext.database.jvm.ProJdbcConnection.open(Unknown Source)
    at liquibase.database.ConnectionServiceFactory.create(ConnectionServiceFactory.java:33)
    ... 17 more
```

The database is up, and I can connect to it from my local OS. My expectation was that I would be able to make the Liquibase container connect to it via the default network. Can anyone help me? I don't know what I'm missing here.

**Comment by Stonezdev** [comment]
https://github.com/liquibase/docker/issues/28 
Maybe this works for you as well?

**Comment by rscarrasco** [comment]
I tried

```
docker run --rm \
          -e "TZ=UTC" \
           --volume /local/path/to/changelogs:/liquibase/changelog \
           liquibase/liquibase \
           --url='jdbc:sqlserver://0.0.0.0:1433;database=MySandbox;' \
           --changeLogFile=/liquibase/changelog/db-changelog-master.xml \
           --username=sa \
           --password=MyStrongPassword@ \
           --logLevel=debug \
           validate
```

But I've got the same error. I also accessed the container to check Liquibase version, and it  is 4.0.0. I'm not sure that I've got the instructions right. Am I missing something?

**Comment by r2liquibase** [comment]
Please `docker inspect` on your SQL Server container and use that IP address when you run the Liquibase docker container. Let us know how that works.

**Comment by r2liquibase** [comment]
Hi, @rscarrasco ! Did you get this resolved?

**Comment by rscarrasco** [comment]
I'm sorry for the delay. I'll check it up and get back soon enough.

**Comment by rscarrasco** [comment]
It worked! Actually, now it's is complaining about using absolute path, but this is another issue. The original question is solved. I thought that the IP 0.0.0.0 informed by `docker ps` was the right one, but if I've got this right, since there are two containers (liquibase and sql server) trying to communicate, I have to use their network IP -- which I did not knew that could be different.

Thanks @r2liquibase!!
